packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,350 @@
+      type: "error",
+      description:
+        "need to increase number of users or reduce number of variations.",
+    };
+    return mdeResults;
+  }
+  const sigma2 = v / nA;
+  const a_star = 1 - (z ** 2 * sigma2) / mean ** 2;
+  const b_star = -2 * mean;
+  const c_star = mean ** 2 - z ** 2 * sigma2;
+  const disc = b_star ** 2 - 4 * a_star * c_star;
+  const sol_1 = (-b_star + Math.sqrt(disc)) / (2 * a_star);
+  //const sol_2 = (-b_star - Math.sqrt(disc)) / (2 * a_star);
+  const mdeResults: MDEResults = {
+    type: "success",
+    mde: (sol_1 - mean) / mean,
+  };
+  return mdeResults;
+}
+
+export function powerMetricWeeks(
+  powerSettings: PowerCalculationParams,
+): PowerCalculationResults {
+  const metrics = powerSettings.metrics;
+  const sampleSizeAndRuntimeNumeric: number[] = []; //for each metric, the first week they achieve 80% power.
+  const nWeeks = powerSettings.nWeeks;
+  let sequentialTuningParameter = 0.0;
+  if (powerSettings.statsEngine.sequentialTesting !== false) {
+    sequentialTuningParameter = powerSettings.statsEngine.sequentialTesting;
+  }
+  function getNumberOfMetrics(params: PowerCalculationParams): number {
+    return Object.keys(params.metrics).length;
+  }
+  const nMetrics = getNumberOfMetrics(powerSettings);
+  const metricKeys = Object.keys(metrics);
+  const mySampleSizeAndRuntime: { [id: string]: SampleSizeAndRuntime | undefined } = {};
+
+  const metricThresholds = {};
+  const weeks: Week[] = [...Array(nWeeks).keys()].map((idx) => ({
+    users: (idx + 1) * powerSettings.usersPerWeek,
+    metrics: {},
+  }));
+
+  for (let i = 0; i < nMetrics; i++) {
+    const metricKey = metricKeys[i];
+    const thisMetric = metrics[metricKey];
+    let thisMean = 0;
+    let thisVariance = 1;
+    if (thisMetric.type === "binomial") {
+      thisMean = thisMetric.conversionRate;
+      thisVariance =
+        thisMetric.conversionRate * (1 - thisMetric.conversionRate);
+    } else {
+      thisMean = thisMetric.mean;
+      thisVariance = thisMetric.standardDeviation ** 2;
+    }
+    let thisMDENumeric = NaN;
+    let thisSampleSizeAndRuntimeNumeric = 999;
+    let lookingForSampleSizeAndRunTime = true;
+    for (let j = 0; j < nWeeks; j++) {
+      const n = powerSettings.usersPerWeek * (j + 1);
+      const thisPower = powerEst(
+        thisMetric.effectSize,
+        thisMean,
+        thisVariance,
+        n,
+        powerSettings.nVariations,
+        powerSettings.alpha,
+        true,
+        sequentialTuningParameter,
+      );
+      if (thisPower >= 0.8 && lookingForSampleSizeAndRunTime) {
+        lookingForSampleSizeAndRunTime = false;
+        thisSampleSizeAndRuntimeNumeric = j + 1;
+      }
+      const thisMde = findMde(
+        0.8,
+        thisMean,
+        thisVariance,
+        powerSettings.usersPerWeek * (j + 1),
+        powerSettings.nVariations,
+        powerSettings.alpha,
+        sequentialTuningParameter,
+      );
+      if (thisMde.type === "error") {
```

Comment:
I'd love to see screenshots of when this error comes up and how we're handling it. I know we discussed not disrupting the main results page when there was no MDE and handling it somewhat gracefully but it's a little tough to see without spending a ton of time following all the code around.

Reply:
attached is a screenshot when nUsers = 10
<img width="1728" alt="Screenshot 2024-04-25 at 3 54 23 PM" src="https://github.com/growthbook/growthbook/assets/24700956/8973994b-8d54-4c27-a302-e036e24cd20f">

Reply:
Hmm, I thought we agreed on filling in like N/A or something for the times when the MDE was not attainable? Then you could show the power as 0% the whole time. That way we could also more gracefully handle cases where the MDE was not defined in the early period but was defined with more data.

Reply:
agreed, and now implemented (PFA screenshot)
<img width="1728" alt="Screenshot 2024-04-26 at 2 43 48 PM" src="https://github.com/growthbook/growthbook/assets/24700956/e46da794-7d5b-41ed-9ddd-90857e54f48f">